### PR TITLE
DUPLO-10598: k8s jobs

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
# k8s Jobs

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-10598 TF: K8s Jobs + Cronjobs Support](https://app.clickup.com/t/8655600/DUPLO-10598)

**Review by joek-duplo — COMMENTED** [review]
LGTM

File: duplocloud/resource_duplo_k8s_job.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[TRACE] resourceKubernetesJobV1Delete(%s, %s): start", tenantId, name)
+
+	// Get the object from Duplo, detecting a missing object
+	c := meta.(*duplosdk.Client)
+	rp, err := c.K8sJobGet(tenantId, name)
+	if err != nil {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8s_job.go:204]: If the backend API error code is 404 on delete, Terraform should successfully destroy the resource.

File: duplocloud/resource_duplo_k8s_job.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	// initiate create Job
+	c := meta.(*duplosdk.Client)
+	err = c.K8sJobCreate(tenantID, &rq)
+	if err != nil {
+		return diag.Errorf("Failed to create Job! API error: %s", err)
+	}
+
+	// wait for completion
+	id := fmt.Sprintf("v3/subscriptions/%s/k8s/job/%s", tenantID, name)
+	diags := waitForResourceToBePresentAfterCreate(ctx, d, "k8s job", id, func() (interface{}, duplosdk.ClientError) {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8s_job.go:92]: Are we not waiting for k8s job status "complete"?

File: duplocloud/resource_duplo_k8s_job.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+			Type:        schema.TypeList,
+			Description: "Spec of the job owned by the cluster",
+			Required:    true,
+			MaxItems:    1,
+			ForceNew:    false,
+			Elem: &schema.Resource{
+				Schema: jobSpecFields(false),
+			},
+		},
+		"wait_for_completion": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8s_job.go:55]: Are we not using wait_for_completion flag for wait ?

File: duplocloud/resource_duplo_k8s_job.go (hunk 1 above)
Comment (joek-duplo) [reply duplocloud/resource_duplo_k8s_job.go:204]: Yes, @tahir-duplo is right.  This also reduces support requests from customers.

File: duplocloud/resource_duplo_k8s_job.go (hunk 2 above)
Comment (joek-duplo) [reply duplocloud/resource_duplo_k8s_job.go:92]: We definitely need an optional feature for that. @ddiaz-duplo can add that?  Should be easy right?

File: duplocloud/resource_duplo_k8s_job.go (hunk 3 above)
Comment (joek-duplo) [reply duplocloud/resource_duplo_k8s_job.go:55]: We need to

**Review by joek-duplo — CHANGES_REQUESTED** [review]
See @tahir-duplo 's comments

File: duplocloud/resource_duplo_k8s_job.go (hunk 2 above)
Comment (ddiaz-duplo) [reply duplocloud/resource_duplo_k8s_job.go:92]: Should we wait for a job complete status? I would think that important thing would be to wait for it to be available to query or view in the UI.

Job complete could take many minutes depending on the type of workload.

File: duplocloud/resource_duplo_k8s_job.go (hunk 3 above)
Comment (ddiaz-duplo) [reply duplocloud/resource_duplo_k8s_job.go:55]: ah, didn't see Joes reply. ok i'll implement that

File: duplocloud/resource_duplo_k8s_job.go (hunk 2 above)
Comment (ddiaz-duplo) [reply duplocloud/resource_duplo_k8s_job.go:92]: didn't see Joe's comment, I'll implement it now.

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**